### DISCUSSION TITLE
Question on "trashAssetsBeforeRun" for Screenshots

### DISCUSSION THREAD

**Discussion by ItsWasteD** [cypress-io/cypress#28665] [Questions and Help]
What exactly gets trashed if I set this configuration to true? Is it only image filetypes or everything within the folder specified? I ask this because I use a subfolder of the screenshot folder for a specific command so lets say I have a scheduled run that runs every day that uses "cypress/screenshots" as screenshot folder. Now I also have a run that run twice a day that uses "cypress/screenshots/otherRun" as screenshots folder. So will this folder get deleted every time the daily run executes? Or will only all images be deleted?

**Comment by portyu9** [comment]
Hi @ItsWasteD 

Yes. Cypress deletes the **entire contents** of `screenshotsFolder`, not just screenshot image files.

So if you have:

```text
cypress/
  screenshots/
    otherRun/
      some-file.png
```

then `otherRun/` and everything inside it will also be removed before `cypress run` when the default configuration is used.

The current option name is:

```js
trashAssetsBeforeRuns: true
```

If you need to retain `otherRun`, either:

- store it outside the configured `screenshotsFolder`, which is the safer approach, or
- set `trashAssetsBeforeRuns: false` and manage cleanup yourself.

Also note that this automatic cleanup happens for `cypress run`, not `cypress open`.

Please mark this answer as accepted if it helped, thank you.
